macros.in

```diff
@@ -1006,6 +1006,34 @@ package or when debugging this package.\
 %_target_vendor		%{_host_vendor}
 %_target_os		%{_host_os}
 
+#==============================================================================
+# ---- compiler flags.
+
+# C compiler flags.  This is traditionally called CFLAGS in makefiles.
+# Historically also available as %%{optflags}, and %%build sets the
+# environment variable RPM_OPT_FLAGS to this value.
+%build_cflags %{optflags}
+
+# C++ compiler flags.  This is traditionally called CXXFLAGS in makefiles.
+%build_cxxflags %{optflags}
+
+# Fortran compiler flags.  Makefiles use both FFLAGS and FCFLAGS as
+# the corresponding variable names.
+%build_fflags %{optflags} %{?_fmoddir:-I%{_fmoddir}}
+
+# Link editor flags.  This is usually called LDFLAGS in makefiles.
+#%build_ldflags -Wl,-z,relro %{?_lto_cflags}
+
+# Expands to shell code to seot the compiler/linker environment
+# variables CFLAGS, CXXFLAGS, FFLAGS, FCFLAGS, LDFLAGS if they have
+# not been set already.
+%set_build_flags \
+  CFLAGS="${CFLAGS:-%{build_cflags}}" ; export CFLAGS ; \
```

Comment:
since we are moving this to RPM, I think it is better to go directly with `export CFLAGS=…`. It is very very old version of bash which did not support this. I think even RHEL6 has good bash enough.

Reply:
Well, but rpm only ever uses /bin/sh. We can't assume bash.

Reply:
https://pubs.opengroup.org/onlinepubs/009695399/utilities/export.html

POSIX says:

```
IEEE Std 1003.1-2001/Cor 1-2002, item XCU/TC1/D6/6 is applied, adding the following text to the end of the first paragraph of the DESCRIPTION: ``If the name of a variable is followed by = word, then the value of that variable shall be set to word.''. The reason for this change is that the SYNOPSIS for export includes:

export name[=word]...

but the meaning of the optional ``= word'' is never explained in the text.
```

So I think only non-POSIX conformant shells use `foo=bar; export foo`

Reply:
Whatever. There's just no real benefit to making it potentially more incompatible.